kedro-datasets/kedro_datasets/__init__.py

```diff
@@ -1,3 +1,17 @@
 """``kedro_datasets`` is where you can find all of Kedro's data connectors."""
 
+__all__ = ["KedroDeprecationWarning"]
 __version__ = "1.7.0"
+
+import warnings
+
+try:
+    # Custom `KedroDeprecationWarning` class was added in Kedro 0.18.14.
+    from kedro import KedroDeprecationWarning
+except ImportError:
+
+    class KedroDeprecationWarning(DeprecationWarning):
+        """Custom class for warnings about deprecated Kedro features."""
+
+
+warnings.filterwarnings("default", category=KedroDeprecationWarning)
```

Comment:
Since the global warning registry overrides any CLI option https://discuss.python.org/t/how-to-silence-pkg-resources-warnings/28629/7?u=astrojuanlu, could we have this behind a conditional?

```suggestion
if not sys.warnoptions:
    warnings.filterwarnings("default", category=KedroDeprecationWarning)
```

see also https://docs.python.org/3.7/library/warnings.html#overriding-the-default-filter

Reply:
As discussed over Slack, I will raise a small PR to do this after in order to avoid being blocked by/battling the CI once again for this large PR. :)

Reply:
I have bad news...

![image](https://github.com/kedro-org/kedro-plugins/assets/316517/c64d57bf-1eb4-475f-8a96-74d7d402c4d6)

Reply:
I was hoping it wouldn't be necessary, but I guess I'll go ahead and make the change in this PR now that it's come to this.